fe/src/main/java/org/apache/doris/alter/MaterializedViewHandler.java

```diff
@@ -99,6 +99,11 @@ public MaterializedViewHandler() {
      */
     public void processCreateMaterializedView(CreateMaterializedViewStmt addMVClause, Database db, OlapTable olapTable)
             throws DdlException, AnalysisException {
+
+        if (olapTable.hasTempPartition(null)) {
```

Comment:
This check is strange. we could add a `existTempPartition` method to olapTable, and  add a `isEmpty` method to 
TempPartitions.